haystack/schema.py

```diff
@@ -258,4 +260,21 @@ def run(self, *args: Any, **kwargs: Any):
         :param kwargs:
         :return:
         """
-        pass
\ No newline at end of file
+        pass
+
+    def set_pipeline_config(self, **kwargs):
+        """
+        Save the init parameters of a component that later can be used with exporting
+        YAML configuration of a Pipeline.
+
+        :param kwargs: all parameters passed to the __init__() of the Component.
+        """
+        if not self.pipeline_config:
+            self.pipeline_config = {"params": {}, "type": type(self).__name__}
+            parameters = inspect.signature(type(self)).parameters
+            for k, v in kwargs.items():
+                if parameters[k].default != v:
```

Comment:
Please add a comment that this ensures to only store non-default values. 

I am also not sure if there won't be cases where we need even those defaults. It's not the case for the YAML export, but I can imagine cases where we just want to access all current params (e.g. experiment tracking) and don't want to manually determine the defaults from `inspect.signature()`. Would it be an option to store them all here and add a `get_pipeline_config(return_defaults=True)` that does the optional filtering of defaults?

Reply:
> Would it be an option to store them all here and add a get_pipeline_config(return_defaults=True) that does the optional filtering of defaults?

Good idea! I added `return_defaults` flag in `save_to_yaml()`.